### PR TITLE
Implement the long-click feature.

### LINKED ISSUE #11138: Long-click a tag to add a subtag
**Is your feature request related to a problem? Please describe.**

#10966 has added support for hierarchical tag (which should be merged soon). In the tag picker dialog, user may want to long-click a tag to add a subtag for it.

**Describe the solution you'd like**

Open the add tag dialog with `$current_tag + ::` as the initial input.

### PR DESCRIPTION
**Pull request by RogerDTZ**
## Purpose / Description
Long-click a tag in the tag picker dialog now opens the add tag dialog with `$current_tag + ::` as the prefilled input.

* Long-click a tag in `FILTER_BY_TAG` dialog does nothing.
* Long-click a tag in `EDIT_TAGS` dialog opens the add tag dialog.

## Fixes
Fixes #11138 

## Approach
* Set `onLongClickListener` to open the add tag dialog for each tag item in `TagsArrayAdapter`.
* The real tag content is passed by setting `View.tag`.
* Prefill the prefix tag by `EditText.setText()`.

## How Has This Been Tested?

Tested on a physical device, Huawei P40 Pro.
* Long-click a tag in `EDIT_TAGS` dialog opens the add tag dialog:
  1. In note editor, open the tag dialog.
  2. Long-click an existing tag `root`.
  3. The add tag dialog shows up, with `root::` prefilled.
* Long-click a tag in `FILTER_BY_TAG` dialog does nothing:
  1. In Card browser, choose "Filter by tag" to open the tag dialog.
  2. Long-click an existing tag `root`.
  3. No add tag dialog jumps up. The clicked item acts as if it is short clicked (toggle between expanded/collapsed or checked / unchecked)

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.kt [hunk 1]
```diff
@@ -214,10 +214,23 @@ class TagsArrayAdapter(private val tags: TagsList, private val resources: Resour
      */
     private val mTagToIsExpanded: HashMap<String, Boolean>
 
+    /**
+     * Long click listener for each tag item. Used to add a subtag for the clicked tag.
+     */
+    private var mTagLongClickListener: View.OnLongClickListener? = null
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.kt:220]: (optional): feel free to get rid of the `m` and change this to a property.

**Review by david-allison — APPROVED** [review]
Looks great! Could you drop a few `@NeedsTest` annotations for a future developer to know what we'd like to test here

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.kt (hunk 1 above)
Comment (RogerDTZ) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsArrayAdapter.kt:220]: Sure!

**Comment by RogerDTZ** [comment]
I added another test to clear one of the old `@NeedsTest`. Haven't figure out an elegant way to test the new feature.

**Review by mikehardy — APPROVED** [review]
Nice! thank you

**Pull request merged**